src/zjs_spi.c

```diff
@@ -171,15 +177,19 @@ static ZJS_DECL_FUNC(zjs_spi_transceive)
                                tx_buf->bufsize, rx_buf->buffer,
                                rx_buf->bufsize) != 0) {
                 jerry_release_value(rx_buf_obj);
+                jerry_release_value(tx_buf_obj);
```

Comment:
This is a good case for using ZVAL_MUTABLE to declare tx_buf_obj above (and iniitalize it to 0). Then it will be automatically freed on all return paths and you can remove four copies of jerry_release_value down here.

The "MUTABLE" part is what allows you to overwrite it in those different paths above. We try to avoid using it because regular ZVAL is safer, but this is the kind of case where you need it. (The immutable version uses `const` underneath so you can't overwrite it.)

You should also be able to use it for rx_buf_obj and initialize that to 0 (represents a 0 integer) instead. (On one path it gets set to jerry_create_null() even though it already had been.)